frontend/src/concepts/areas/types.ts

```diff
@@ -0,0 +1,79 @@
+import { EitherOrBoth } from '~/typeHelpers';
+import { DashboardCommonConfig } from '~/k8sTypes';
+
+/** All areas that we need to support in some fashion or another */
+export enum SupportedArea {
+  /* Standalone areas */
+  DS_PIPELINES = 'ds-pipelines',
+  // TODO: Jupyter Tile Support? (outside of feature flags today)
+  WORKBENCHES = 'workbenches',
+  // TODO: Support Applications/Tile area
+  // TODO: Support resources area
+
+  /* Admin areas */
+  BYON = 'bring-your-own-notebook',
+  CLUSTER_SETTINGS = 'cluster-settings',
+  USER_MANAGEMENT = 'user-management',
+
+  /* DS Projects specific areas */
+  DS_PROJECTS_PERMISSIONS = 'ds-projects-permission',
+  DS_PROJECTS_VIEW = 'ds-projects',
+
+  /* Model Serving areas */
+  MODEL_SERVING = 'model-serving-shell',
+  CUSTOM_RUNTIMES = 'custom-serving-runtimes',
+  K_SERVE = 'kserve',
+  MODEL_MESH = 'model-mesh',
+}
+
+/** Components deployed by the Operator. Part of the DSC Status. */
+export enum StackComponent {
+  CODE_FLARE = 'codeflare',
+  DS_PIPELINES = 'data-science-pipelines-operator',
+  K_SERVE = 'kserve',
+  MODE_MESH = 'model-mesh',
+  // Bug: https://github.com/opendatahub-io/opendatahub-operator/issues/641
+  DASHBOARD = 'odh-dashboard',
+  RAY = 'ray',
+  WORKBENCHES = 'workbenches',
+}
+
+// TODO: Support extra operators, like the pipelines operator -- maybe as a "external dependency need?"
+type SupportedComponentFlagValue = {
+  /**
+   * An area can be reliant on another area being enabled. The list is "OR"-ed together.
+   *
+   * Example, Model Serving is a shell for either KServe or ModelMesh. It has no value on its own.
+   * It can also be a chain of reliance... example, Custom Runtimes is a Model Serving feature.
+   *
+   * TODO: support AND -- maybe double array?
+   */
+  reliantAreas?: SupportedArea[];
+} & EitherOrBoth<
+  {
+    /**
+     * Refers to OdhDashboardConfig's feature flags, any number of them to be "enabled", the result
+     * is AND-ed. Omit to not be related to any feature flag.
+     *
+     * Note: "disable<FlagName>" methodology is confusing and needs to be removed
+     * Note: "Enabled" will mean "disable<FlagName>" is false
+     * @see https://github.com/opendatahub-io/odh-dashboard/issues/1108
+     */
+    featureFlags: (keyof DashboardCommonConfig)[];
```

Comment:
Doesn't `keyof DashboardCommonConfig` contain more than just feature flags?

Also later on you also accept `notebookController` as a valid feature flag.

Reply:
Yeah, this needs to be better configured -- I'll look into this.